### DISCUSSION TITLE
Enable promoted attributes in the mobile frontend?

### DISCUSSION THREAD

**Discussion by LokiLuciferase** [TriliumNext/Trilium#4314] [Ideas] [answered]
Hi,

My workflows rely heavily on pre-defined promoted attributes inherited from templates, and unfortunately these do not seem to be displayed out of the box on the mobile frontend - making the otherwise nice mobile frontend unusable to me.

I'm not sure it's even possible to add custom widgets to the mobile frontend, but if so, is it possible to somehow enable showing those via a user-defined widget? Or am I missing a user-accessible option somewhere?

Otherwise, would the maintainers be interested in a PR that adds this functionality to the codebase, either permanently or as opt-in by options? Making the requisite code change is trivial but I appreciate that there might be a desire to keep the mobile frontend simple and clean. Although I think editing existing promoted attributes would not detract from this goal much (see attached).

Either way thanks for the amazing software!

![off](https://github.com/zadam/trilium/assets/22958522/6bb194f7-496c-4b93-9f07-a1fd1b4016e8)
![on](https://github.com/zadam/trilium/assets/22958522/b8778873-4a86-4b16-919e-425485b2a97c)

**Comment by zerebos** [comment]
I agree it wouldn't detract from mobile experience to have those even as an option. I personally think the mobile frontend could use some love overall and it's something on my to-do list to explore as I get more comfortable with this codebase.

**Comment by zadam** [comment, marked as answer]
Hi, I've added promoted attributes widget to the mobile version into master (future 0.62).

There's one issue - the promoted relations are readonly because the autocomplete library is not present in the mobile version (not sure, but likely this was done to keep the mobile version light).

**Reply by LokiLuciferase** [reply]
Thanks so much!
